### DISCUSSION TITLE
Got expected schema argument to be a string, got <class 'sqlalchemy.engine.base.Engine'>

### DISCUSSION THREAD

**Discussion by eniolw** [agronholm/sqlacodegen#333] [Q&A]
After I run the equivalent to this:

sqlacodegen mssql+pymssql://user:pass@localhost/db > models.py

I get this:

sqlalchemy.exc.ArgumentError: expected schema argument to be a string, got <class 'sqlalchemy.engine.base.Engine'>.

I'm using
Python 3.10
SQLAlchemy==2.0.30
sqlacodegen==2.3.0
pymssql==2.3.0
ODBC Driver 18 for SQL Server

Ultimately, how do I get the models from this db that is in sql server using sqlacodegen?

Thanks.

**Comment by agronholm** [comment]
Have you tried with the latest release candidate?

**Comment by Yiu-Kelvin** [comment]
I had this exact problem, but its fixed after installing a newer version of sqlacodegen:
`pip install sqlacodegen==3.0.0rc2`
